WIP: pr/land: Verify that a change can be landed

abhinav opened the pull request:
This updates `git pr land` to verify that a change passes all checks
before landing. This includes verifying that there are no blocking
reviewers and all builds are passing.

Resolves #24

abhinav committed:
pr/land: Verify that a change can be landed

This updates `git pr land` to verify that a change passes all checks
before landing. This includes verifying that there are no blocking
reviewers and all builds are passing.

Resolves #24

abhinav committed:
Allow skipping landing checks with --no-check

abhinav committed:
changelog

prashantv commented on pr/land.go:
```diff
@@ -2,14 +2,103 @@ package pr
+			continue
+		}
+
+		err = multierr.Append(err,
```
I think you want the last state for a user, since the same user can have multiple reviews, 1 where they asked for changes and then a later one where they accept. It looks like the API does return all of them, so you'll want to maintain a map of user id -> lastState

abhinav committed:
Only consider the most recent review

abhinav committed:
Test landCmd

abhinav committed:
Pull Request Review State: DISMISSED

abhinav committed:
Extract approval check into Review service

A PR is now landable if it has at least one approval and no blocking
reviews or failing checks.
